### ISSUE TITLE
CLI: cannot bind graph symbols before analysis, and broadcast(sym, sym) does not simplify

### ISSUE THREAD

**Issue by czoli1976** [sonos/tract#2650]
Two separate obstacles hit while importing an ORT-GenAI decoder export (`onnx-community/Bonsai-1.7B-ONNX`, Qwen3), both about binding the graph's symbolic dimensions to concrete ones for a single decode or prefill step.

## 1. `--set` runs too late to help the analyse stage

`--set` is documented as "set a symbol to a concrete value after decluttering", and the analyse stage runs well before that. The export computes `seqlens_k` from `Shape(attention_mask)`, and analysis fails there:

```
Error at stage "analyse"
  1: Failed analyse for node #594 "/model/attn_mask_reformat/attn_mask_subgraph/Expand" MultiBroadcastTo
  3: Applying rule outputs[0].shape == 1,1
  4: Unifying shapes batch_size,1 and 1,1
  5: Impossible to unify Sym(batch_size) with Val(1).
```

`--input-facts-from-bundle` sets the input facts but leaves the declared symbols in place, so part of the graph sees `batch_size` and part sees `1`. There appears to be no way to bind a symbol before analysis from the CLI.

The workaround is to rewrite the model, replacing `dim_param` with `dim_value` on every input, output and `value_info`. That works, but it means the CLI cannot run a symbolic decoder export as shipped.

## 2. A runtime scalar that feeds a shape leaves an unresolvable symbol

These exports take `num_logits_to_keep` as an `int64` scalar *input* and use it to `Slice` the hidden states before the `lm_head`, so the sequence dim of that matmul is only known at runtime. tract models it as an anonymous symbol, which then fails to reconcile with the declared output shape:

```
Error at stage "type"
  0: Translating node #1142 "/lm_head/MatMul_Quant" MatMulNBits ToTypedTranslator
  1: Output mismatch after rewiring expansion for output #0:
     expected 1,8,151669,F32 got 1,broadcast((<Sym0>), (<Sym0>)),151669,F32
```

Two things stand out. `--constantize num_logits_to_keep` does not resolve it. And `broadcast((<Sym0>), (<Sym0>))` is a broadcast of a symbol with itself, which should simplify to that symbol regardless of the rest — that alone may be worth fixing, since it would make the shape reconcile in this case.

## How to fix

For (1), the clean fix is a pre-analysis binding: either let `--set` (or a sibling flag) apply to the inference model before analysis, or have `--input-facts-from-bundle` substitute the symbols it can infer from the bundle's concrete shapes rather than only setting input facts. Either removes the need to rewrite the file.

For (2), simplifying `broadcast(a, a)` to `a` in the `TDim` broadcast constructor is a contained change and would let the shapes unify here; whether the anonymous symbol should also be bindable by `--constantize` is a separate question.

Workaround meanwhile, for both: freeze `dim_param` to `dim_value` in the model, and turn `num_logits_to_keep` into an initializer. With that done the model imports and runs, and its logits match onnxruntime to 1e-4 absolute.
